### PR TITLE
Protocol: Use correct calculation for delta movements

### LINKED ISSUE #5030: Mobs Y position glitch - 1.8.9
Client version: 1.8.9
Server OS: Windows 10 Pro
Cuberite Commit id: 14a00d0051d9c0f97105145bd2ecf7679bb015a3

### Expected behavior
Mobs walking on the floor normally.

### Actual behavior
When using 1.8.9, Mobs Y axis gets crazy and makes them float. I've just tested in 1.8.9 and 1.12, It just happened with 1.8.9. Me and a friend were seeing the mobs at the same height, but it wouldn't happen with a newer minecraft version. Another thing to note is that horses that were floating became unridable, when interacted, it resets your camera position. 

### Steps to reproduce the behavior
Use 1.8.9 Vanilla and just walk thru the map watching the mobs as they will **somehow** render floating.

### Server log
Nothing was shown in the log, as it was client-sided.

### PR DESCRIPTION
**Pull request by tigerw**
* Fixes #5030.

### CODE REVIEW

**Review by bearbin — COMMENTED** [review]
This code will now send out lots more almost-0 broadcasts. Is that desirable?

Otherwise looks fine.

**Comment by tigerw** [comment]
GitHub Desktop has PUSH NOTIFICATIONS what a time to be alive!!

**Comment by tigerw** [comment]
> This code will now send out lots more almost-0 broadcasts. Is that desirable?

I will do a test with a piggy and report back.

File: src/Protocol/Protocol_1_8.cpp [hunk 1]
```diff
@@ -579,12 +579,14 @@ void cProtocol_1_8_0::SendEntityPosition(const cEntity & a_Entity)
...
-
-	// Limitations of a byte
-	static const auto Max = std::numeric_limits<Int8>::max();
-
-	if ((std::abs(Delta.x) <= Max) && (std::abs(Delta.y) <= Max) && (std::abs(Delta.z) <= Max))
+	const auto Delta = (a_Entity.GetPosition() * 32).Floor() - (a_Entity.GetLastSentPosition() * 32).Floor();
+
+	// Ensure that the delta is within range of a BEInt8:
+	if (
+		(-128 <= Delta.x) && (Delta.x <= 127) &&
```
[the remark this answers is not in the document]
Comment (tigerw) [reply src/Protocol/Protocol_1_8.cpp:586]: In the spirit of cpp our Int8 "may" have a range of -127 to 127, so what about a constant instead, called BEInt8Min/Max?

**Comment by 12xx12** [comment]
I agree with @bearbin. The magic numbers are confusing and the numeric limits should be preferred. Else lgtm

File: src/Protocol/Protocol_1_8.cpp (hunk 1 above)
[the remark this answers is not in the document]
Comment (bearbin) [reply src/Protocol/Protocol_1_8.cpp:586]: Seems like a good idea, it can go in one of the protocol base headers?

**Comment by tigerw** [comment]
Pig test subject
Fig. 1: Unwitting/unwilling test subject.

On 1.8: 33773 calls to SendEntityPosition had a non-zero delta, out of a total of 35932 calls, which is 94%.
On 1.12.2: 30039 calls had a non-zero delta, out of 30499 in total, which is 98.5%.

In this case I would propose to send a Teleport whenever the delta was zero to keep server and client in sync (as opposed to delaying the packet as before), since plugins may want to send small movements and expect it to update on the client.

**Review by bearbin — APPROVED** [review]

**Pull request merged**